internal/eval/eval.go

```diff
@@ -2883,6 +2883,26 @@ func evalHSET(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(numKeys, false)
 }
 
+func evalHSETNX(args []string, store *dstore.Store) []byte {
+	if len(args) != 3 {
+		return diceerrors.NewErrArity("HSETNX")
+	}
+
+	key := args[0]
+	hmKey := args[1]
+
+	val, errWithMessage := getValueFromHashMap(key, hmKey, store)
+	if errWithMessage != nil {
+		return errWithMessage
```

Comment:
Error handling needs to be improved. Instead of directly returning errWithMessage, it would be better to wrap it with additional context for better readability and debugging. You can use `diceerrors.NewErrWithFormattedMessage(errWithMessage)`.

For example:
```go
errWithMessage := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
if errWithMessage != nil {
    return diceerrors.NewErrWithFormattedMessage(errWithMessage)
}

Reply:
The behavior needs to line up exactly with how redis would respond.

Reply:
@manishrw the `getValueFromHashMap` function already returns the error byte wrapped with `diceerrors.NewErrWithFormattedMessage` [here](https://github.com/DiceDB/dice/blob/5efae7ae6aec5641b85cf4c9db3cea2e09dc8310/internal/eval/hmap.go#L81).  Is further change needed here?

Attaching the current behaviour for reference (7379 is dice, 8989 is redis-server) :- 
![image](https://github.com/user-attachments/assets/dde68f71-3f00-444d-b980-86bf37b5a4f0)

Reply:
No change is needed.